### PR TITLE
api: continued improvements

### PR DESCRIPTION
**Pull request by sanderpick**
After using the API to build a shell, I realized it's not very good yet. Here are some tweaks to help.

### CODE REVIEW

**Comment by sanderpick** [comment]
This pairs with https://github.com/textileio/go-textile-threads/pull/29

File: threadservice/options.go [hunk 1]
```diff
@@ -114,29 +114,29 @@ func PutOptions(opts ...PutOption) *PutSettings {
 	return options
 }
 
-// ListenOpt is an instance helper for creating listen options.
-var ListenOpt ListenOption
+// SubOpt is an instance helper for creating subscription options.
+var SubOpt SubOption
```
Comment (jsign) [inline threadservice/options.go:118]: Is this used for something? (same with `AddOpt`)

**Review by jsign — APPROVED** [review]
LGTM.
Particularly, always nice to see more typed returns (mostly with structured `[]byte` ones).

File: threadservice/options.go (hunk 1 above)
Comment (sanderpick) [reply threadservice/options.go:118]: These are just helper for adding options: https://github.com/textileio/go-textile-threads/blob/master/test/threads_suite.go#L97. There very well could be a better way to handle options in general

**Pull request merged**